op-supervisor: Event overflow on large unsafe head gap

The supervisor can enter a non-recoverable crash-loop when the unsafe head gap between its log DB and a managed node grows too large. This seems to be caused by the supervisor's `ChainProcessor` entering a loop triggered by a `ChainProcessEvent` event that then internally emits `ChainIndexingContinueEvent`. These stack up, eventually leading to an even overflow panic (taken from `dev-infra-rehearsal-0-bn-op-supervisor-snapsync-0`):
```
t=2025-06-09T14:38:12+0000 lvl=debug msg="Building new HazardSet" chain=420120009 worker=cross-unsafe chainID=420120009 block="BlockSeal(hash:0xa1a0991221ad53f62f9cbd608dff4eff0b47ea2a27591e304b4a94c27a4d8b1e, number:125630, time:1749236224)"
t=2025-06-09T14:38:12+0000 lvl=debug msg="Processing block for hazards" chain=420120009 worker=cross-unsafe chainID=420120009 block="BlockSeal(hash:0xa1a0991221ad53f62f9cbd608dff4eff0b47ea2a27591e304b4a94c27a4d8b1e, number:125630, time:1749236224)"
t=2025-06-09T14:38:12+0000 lvl=debug msg="Successfully built HazardSet" chain=420120009 worker=cross-unsafe chainID=420120009 block="BlockSeal(hash:0xa1a0991221ad53f62f9cbd608dff4eff0b47ea2a27591e304b4a94c27a4d8b1e, number:125630, time:1749236224)"
t=2025-06-09T14:38:12+0000 lvl=info msg="Updated cross-unsafe" chain=420120009 crossUnsafe="BlockSeal(hash:0xa1a0991221ad53f62f9cbd608dff4eff0b47ea2a27591e304b4a94c27a4d8b1e, number:125630, time:1749236224)"
t=2025-06-09T14:38:12+0000 lvl=debug msg="Building new HazardSet" chain=420120010 worker=cross-unsafe chainID=420120010 block="BlockSeal(hash:0x9cfcc8f132f5467055073e18d0230eac29c7771e107b1d196442319cae65a0bb, number:241533, time:1749226497)"
t=2025-06-09T14:38:12+0000 lvl=debug msg="Processing block for hazards" chain=420120010 worker=cross-unsafe chainID=420120010 block="BlockSeal(hash:0x9cfcc8f132f5467055073e18d0230eac29c7771e107b1d196442319cae65a0bb, number:241533, time:1749226497)"
t=2025-06-09T14:38:12+0000 lvl=debug msg="Successfully built HazardSet" chain=420120010 worker=cross-unsafe chainID=420120010 block="BlockSeal(hash:0x9cfcc8f132f5467055073e18d0230eac29c7771e107b1d196442319cae65a0bb, number:241533, time:1749226497)"
t=2025-06-09T14:38:12+0000 lvl=info msg="Updated cross-unsafe" chain=420120010 crossUnsafe="BlockSeal(hash:0x9cfcc8f132f5467055073e18d0230eac29c7771e107b1d196442319cae65a0bb, number:241533, time:1749226497)"
t=2025-06-09T14:38:12+0000 lvl=error msg="Failed to enqueue event" emitter=chainsDBs event=cross-unsafe-update context=0
panic: something is very wrong, queued up too many events! Dropping event "cross-unsafe-update"
goroutine 44 [running]:
github.com/ethereum-optimism/optimism/op-node/rollup/event.(*Sys).emit(0xc00004e600, {0x11badad, 0x9}, 0x0, {0x15578c0, 0xc000d31270}, 0x0)
	/app/op-node/rollup/event/system.go:293 +0x245
github.com/ethereum-optimism/optimism/op-node/rollup/event.(*systemActor).Emit(0xc00047e6c0, {0x15578c0, 0xc000d31270})
	/app/op-node/rollup/event/system.go:81 +0x65
github.com/ethereum-optimism/optimism/op-node/rollup/event.(*Limiter[...]).Emit(0x1553660, {0x15578c0, 0xc000d31270})
	/app/op-node/rollup/event/limiter.go:38 +0xa2
github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/db.(*ChainsDB).UpdateCrossUnsafe(0xc0001aab00, {0x190a85ca, 0x0, 0x0, 0x0}, {{0x9c, 0xfc, 0xc8, 0xf1, 0x32, ...}, ...})
	/app/op-supervisor/supervisor/backend/db/update.go:180 +0x2e3
github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/cross.CrossUnsafeUpdate({0x156f998, 0xc00048e5a0}, {0x190a85ca, 0x0, 0x0, 0x0}, {0x1568bc8, 0xc0001aab00}, {0x15587c0, 0xc000449f70})
	/app/op-supervisor/supervisor/backend/cross/unsafe_update.go:77 +0x982
github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/cross.(*CrossUnsafeWorker).OnEvent(0xc0004c2140, {0x1558920?, 0xc000f90ac0?})
	/app/op-supervisor/supervisor/backend/cross/unsafe_update.go:93 +0x99
github.com/ethereum-optimism/optimism/op-node/rollup/event.(*systemActor).RunEvent(0xc00047eba0, {{0x1558920?, 0xc000f90ac0?}, 0x41c8?, 0x0?})
	/app/op-node/rollup/event/system.go:101 +0x144
github.com/ethereum-optimism/optimism/op-node/rollup/event.(*globalHandle).onEvent(...)
	/app/op-node/rollup/event/executor_global.go:252
github.com/ethereum-optimism/optimism/op-node/rollup/event.(*GlobalSyncExec).processEvent(0xc00060cb60?, {{0x1558920?, 0xc000f90ac0?}, 0x2be2?, 0x0?})
	/app/op-node/rollup/event/executor_global.go:155 +0xeb
github.com/ethereum-optimism/optimism/op-node/rollup/event.(*GlobalSyncExec).Drain(0xc00060cb60)
	/app/op-node/rollup/event/executor_global.go:189 +0x29
github.com/ethereum-optimism/optimism/op-supervisor/supervisor.(*SupervisorService).initBackend.func1()
	/app/op-supervisor/supervisor/service.go:87 +0x25
github.com/ethereum-optimism/optimism/op-service/tasks.(*Poller).Start.func1()
	/app/op-service/tasks/poll.go:57 +0xa4
created by github.com/ethereum-optimism/optimism/op-service/tasks.(*Poller).Start in goroutine 1
	/app/op-service/tasks/poll.go:49 +0x156
```

We need to revisit the event flows here to make sure the supervisor can make incremental progress without events stacking too high. In this particular case I suggest we remove the `ChainIndexingContinueEvent` event completely, as it's just used for the chain processor to communicate with itself, and refactor it instead to run a classic outer loop.

--- a/op-supervisor/supervisor/backend/status/status.go
+++ b/op-supervisor/supervisor/backend/status/status.go
@@ -82,7 +82,8 @@ func (su *StatusTracker) OnEvent(ctx context.Context, ev event.Event) bool {
 	return true
 }
 
-func (su *StatusTracker) HasInitializedStatuses() bool {
+// hasInitializedStatuses is not behind a lock, because it is used only internally
+func (su *StatusTracker) hasInitializedStatuses() bool {
 	for _, nodeStatus := range su.statuses {
 		if nodeStatus != nil && *nodeStatus != (NodeSyncStatus{}) {
 			return true
@@ -97,7 +98,7 @@ func (su *StatusTracker) SyncStatus() (eth.SupervisorSyncStatus, error) {
 
 	// after supervisor restarts, there is a timespan where all node's sync status is not fetched yet
 	// error immediately until at least single node sync status is available, which is not empty
-	if !su.HasInitializedStatuses() {
+	if !su.hasInitializedStatuses() {
 		return eth.SupervisorSyncStatus{}, ErrStatusTrackerNotReady
 	}
 

--- a/op-supervisor/supervisor/backend/status/status.go
+++ b/op-supervisor/supervisor/backend/status/status.go
@@ -83,9 +83,6 @@ func (su *StatusTracker) OnEvent(ctx context.Context, ev event.Event) bool {
 }
 
 func (su *StatusTracker) HasInitializedStatuses() bool {
-	su.mu.RLock()
-	defer su.mu.RUnlock()
-
 	for _, nodeStatus := range su.statuses {
 		if nodeStatus != nil && *nodeStatus != (NodeSyncStatus{}) {
 			return true

